core/src/main/java/com/segment/analytics/kotlin/core/platform/plugins/logger/LogTarget.kt

```diff
@@ -0,0 +1,238 @@
+package com.segment.analytics.kotlin.core.platform.plugins.logger
+
+import com.segment.analytics.kotlin.core.Analytics
+import com.segment.analytics.kotlin.core.BaseEvent
+import com.segment.analytics.kotlin.core.platform.Plugin
+import java.util.*
+
+/**
+ * The foundation for building out a special logger. If logs need to be directed to a certain area, this is the
+ * interface to start off with. For instance a console logger, a networking logger or offline storage logger
+ * would all start off with LogTarget.
+ */
+interface LogTarget {
+
+    /**
+     * Implement this method to process logging messages. This is where the logic for the target will be
+     * added. Feel free to add your own data queueing and offline storage.
+     * - important: Use the Segment Network stack for Segment library compatibility and simplicity.
+     */
+    fun parseLog(log: LogMessage)
+
+    /**
+     * Optional method to implement. This helps respond to potential queueing events being flushed out.
+     * Perhaps responding to backgrounding or networking events, this gives a chance to empty a queue
+     * or pump a firehose of logs.
+     */
+    fun flush()
+}
+
+/**
+ * Used for analytics.log() types. This lets the system know what to filter on and how to set priorities.
+ */
+enum class LogFilterKind {
+    ERROR,
+    WARNING,
+    DEBUG;
+
+    override fun toString(): String {
+        return when(this) {
+            ERROR -> "ERROR"        // Not Verbose (fail cases | non-recoverable errors)
+            WARNING -> "Warning"    // Semi-verbose (deprecations | potential issues)
+            DEBUG -> "Debug"        // Verbose (everything of interest)
+        }
+    }
+}
+
+/**
+ * The Segment logging system has three types of logs: log, metric and history. When adding a target that
+ * responds to logs, it is possible to adhere to 1 to many. In other words, a LoggingType can be .log &
+ * .history. This is used to tell which targets logs are directed to.
+ */
+class LoggingType(types: List<Filter>) {
+
+    enum class Filter {
+        LOG,
+        METRIC,
+        HISTORY;
+    }
+
+    companion object {
+        /// Convenience .log logging type
+        val log = LoggingType(listOf(Filter.LOG))
+        /// Convenience .metric logging type
+        val metric = LoggingType(listOf(Filter.METRIC))
+        /// Convenience .history logging type
+        val history = LoggingType(listOf(Filter.HISTORY))
+    }
+
+    // - Private Properties and Methods
+    private val allTypes: List<Filter> = types
+
+    /**
+     * Convenience method to find if the LoggingType supports a particular destination.
+     *
+     * @property destination The particular destination being tested for conformance.
+     */
+    internal fun contains(destination: Filter): Boolean {
+        return this.allTypes.contains(destination)
```

Comment:
does `allTypes` allow duplicate filters/destinations? if not, it's probably better to use `Set` instead of `List` for better performance.

Reply:
Good call, not just better performance but better api